library/bignum_core.c

```diff
@@ -540,4 +540,18 @@ int mbedtls_mpi_core_get_mont_r2_unsafe( mbedtls_mpi *X,
     return( ret );
 }
 
+void mbedtls_mpi_core_ct_uint_table_lookup( mbedtls_mpi_uint *dest,
+                                        const mbedtls_mpi_uint *table,
+                                        size_t limbs,
+                                        size_t count,
+                                        size_t index )
+{
```

Comment:
While I understand that at this level we trust the module calling this, but I feels is would be nice to have a boundary check for `index <= count-1`.

Reply:
I feel the itch too, but unfortunately this is performance critical code. In the future we will optimise it further and wouldn't want to pay the cost for those extra checks (we would need to return a value and the caller would need to check it).